cmd/root.go

```diff
@@ -799,3 +799,38 @@ func runSignalWrapper(cmd *Command) error {
 	}
 	return err
 }
+
+func quitquitquit(shutdownCh chan<- error) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		if req.Method != http.MethodPost {
+			rw.WriteHeader(400)
+			return
+		}
+		shutdownCh <- errQuitQuitQuit
```

Comment:
What if /quitquitquit was called twice in rapid succession? Would this still exit cleanly?

Reply:
Good point. I'll wrap this in a sync.Once.